### PR TITLE
STM32 :  add soc for stm32u073 series

### PR DESCRIPTION
**Pull request by djiatsaf-st**
select soc for stm32 nucleo_u073xx boards  and irq default configuration

### CODE REVIEW

**Review by nordicjm — DISMISSED** [review]

**Review by erwango — CHANGES_REQUESTED** [review]
Please add u073 packages dtsi files.

**Review by nordicjm — DISMISSED** [review]

File: dts/arm/st/u0/stm32u073XB.dtsi
Comment (erwango) [inline dts/arm/st/u0/stm32u073XB.dtsi:1]: Should be `dts/arm/st/u0/stm32u073Xb.dtsi` according to existing convention.

File: dts/arm/st/u0/stm32u073X8.dtsi [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+		compatible = "zephyr,memory-region", "mmio-sram";
+		reg = <0x20000000 DT_SIZE_K(32)>;
+		zephyr,memory-region = "SRAM1";
+	};
+
+	sram2: memory@20008000 {
+		compatible = "zephyr,memory-region", "mmio-sram";
+		reg = <0x20008000 DT_SIZE_K(8)>;
+		zephyr,memory-region = "SRAM2";
+	};
```
Comment (erwango) [inline dts/arm/st/u0/stm32u073X8.dtsi:21]: Nitpick (non blocking), move the following directly  in stm32u073.dtsi:
```
	sram1: memory@20000000 {
		compatible = "zephyr,memory-region", "mmio-sram";
		zephyr,memory-region = "SRAM1";
	};

	sram2: memory@20008000 {
		compatible = "zephyr,memory-region", "mmio-sram";
		zephyr,memory-region = "SRAM2";
	};
```
and keep the following in this file:
```suggestion
	sram1: memory@20000000 {
		reg = <0x20000000 DT_SIZE_K(32)>;
	};

	sram2: memory@20008000 {
		reg = <0x20008000 DT_SIZE_K(8)>;
	};
```

**Review by erwango — CHANGES_REQUESTED** [review]
We're getting there.
What is missing is to get u083.dtsi including u073.dtsi

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Review by marwaiehm-st — APPROVED** [review]

**Pull request merged**